library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
> I'm not sure I'm following. At which point do we have to worry about open terms? What would be an example tactic invocation?

Even for closed terms, the issue is not trivial.
We would need to convert a `vm_obj` that represents a value of type alpha into a pre-term that when elaborated has type alpha.  Recall that a lot of information is lost at run time. 
I wrote some functions for some types by hand, but I don't see how to make it general enough.
https://github.com/leanprover/lean/blob/master/library/init/meta/tactic.lean#L822-L849

At runtime, a `vm_obj` that represents a value of type alpha is necessarily a closed term.
We have an `vm_expr` `vm_obj` which encapsulates a (potentially open) term and/or pre-term.
This `vm_expr` represents a lean object of type `expr` or `pexpr`.

Now, suppose we have `parse (list name) ident_list`. The action `indent_list` should construct a pre-term that represents a term of type `list name` when elaborated. This `vm_obj` would something of the form `(#1 (vm_name ...) (#1 (vm_name ..) ... #0) ...))`.  We cannot stuck a `vm_obj` in the middle of the abstract syntax tree, even if we use `ts_vm_obj`.  If we do it, we would have to save the object in an .olean file (we don't have support for this). Second, we would not be able to compile lean files with trust level 0, in this setting all macros are erased, and the macro encapsulating the `ts_vm_obj` would be gone. 
We would also have problems with type checking (e.g., what does it mean for two `ts_vm_obj`'s to be definitionally equal?).